src/EnemyBehaviourEC.cpp

```diff
@@ -107,6 +155,34 @@ bool EnemyBehaviourEC::timeToAttack() {
     return false;
 }
 
+Ogre::Vector3 EnemyBehaviourEC::separate() {
+
+    Ogre::Vector3 resultado = Ogre::Vector3(0, 0, 0);
+    int numAgentes = otherTransform.size();
+    for (int i = 0; i < numAgentes; i++) {
+        TransformComponent* agenteObjetivo = otherTransform[i];
+
+        Ogre::Vector3 direccion =
+            dynamic_cast<TransformComponent*>(
+                this->father->getComponent("TransformComponent"))
+                ->getPosition() -
+            agenteObjetivo->getPosition();
+
+        float distancia = direccion.squaredLength();
+
+        if (distancia < separationRadius) {
+            if (distancia == 0.0f)
```

Comment:
Floating-point comparisons are prone to math calculations (e.g. `0.1 + 0.2` doesn't return `0.3`, but rather `0.30000000000000004`). Use a comparison with threshold instead.
```suggestion
            if (distancia < 0.01f)
```